src/ota_image_builder/_common.py

```diff
@@ -134,3 +135,33 @@ def check_if_valid_ota_image(image_root: Path) -> bool:
         logger.debug(f"Resource directory not found: {resource_dir}")
         return False
     return True
+
+
+#
+# ------ NVIDIA Jetson related ------ #
+#
+
+NV_TEGRA_RELEASE_FPATH = "/etc/nv_tegra_release"
+NV_TEGRA_RELEASE_PA = re.compile(
+    (
+        r"# R(?P<major_ver>\d+) \(\w+\), REVISION: (?P<major_rev>\d+)\.(?P<minor_rev>\d+), "
+        r"GCID: (?P<gcid>\d+), BOARD: (?P<board>\w+), EABI: (?P<eabi>\w+)"
+    )
+)
+"""Example:
+
+# R35 (release), REVISION: 4.1, GCID: 33958178, BOARD: t186ref, EABI: aarch64, DATE: Tue Aug  1 19:57:35 UTC 2023
+"""
+
+
+def get_bsp_ver_info(nv_tegra_release_str: str) -> str | None:
+    """Parse BSP version from contents of /etc/nv_tegra_release."""
+    ma = NV_TEGRA_RELEASE_PA.match(nv_tegra_release_str)
+    if not ma:
+        logger.warning(f"invalid nv_tegra_release content: {nv_tegra_release_str}")
+        return
+
+    major_ver = int(ma.group("major_ver"))
+    major_rev = int(ma.group("major_rev"))
+    minor_rev = int(ma.group("minor_rev"))
```

Comment:
Do we need to consider the exceptions?

Reply:
Ah in the above's regex pattern, the `major_ver`, `major_rev` and `minor_rev` capture groups are required, so if the input string doesn't match the pattern, there will be no match and the `ma` object will be None and make `get_bsp_ver_info` returns None before hand.